view_scene.py

```diff
@@ -0,0 +1,112 @@
+"""
+Script to render a GTSFM scene using either Open3d or Mayavi mlab.
+
+Author: John Lambert
+"""
+
+import argparse
+import os
+from pathlib import Path
+
+import numpy as np
+from gtsam import Rot3, Pose3
+
+import gtsfm.utils.io as io_utils
+
+from visualization.open3d_vis_utils import draw_scene_open3d
+from visualization.mayavi_vis_utils import draw_scene_mayavi
+
+REPO_ROOT = Path(__file__).parent.resolve()
+
+
+def compute_point_cloud_center_robust(point_cloud: np.ndarray) -> np.ndarray:
+    """Robustly estimate the point cloud center.
+
+    Args:
+        point_cloud: array of shape (N,3) representing 3d points.
+
+    Returns:
+        mean_pt: coordinates of central point, ignoring outliers.
+    """
+    ranges = np.linalg.norm(point_cloud, axis=1)
+    outlier_thresh = np.percentile(ranges, 75)
+    mean_pt = point_cloud[ranges < outlier_thresh].mean(axis=0)
+    return mean_pt
+
+
+def view_scene(args: argparse.Namespace) -> None:
+    """Read GTSFM output from .txt files and render the scene to the GUI.
+
+    Args:
+        args: rendering options.
+    """
+    points_fpath = f"{args.output_dir}/points3D.txt"
+    images_fpath = f"{args.output_dir}/images.txt"
+    cameras_fpath = f"{args.output_dir}/cameras.txt"
+
+    wTi_list, img_fnames = io_utils.read_images_txt(images_fpath)
+    calibrations = io_utils.read_cameras_txt(cameras_fpath)
```

Comment:
Unrelated to this PR: shouldnt wTi_list be returned by read_cameras semantically?

Reply:
hmm what do you mean by "return semantically"?

Reply:
Is the wTi_list ground truth?

Reply:
no, it just represents poses stored in a COLMAP-format file.
It could be GT, or it could be the output of GTSFM.